tests/components/advantage_air/test_config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Test the Advantage Air config flow."""
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.advantage_air.const import DOMAIN
+
+from tests.async_mock import patch
+from tests.components.advantage_air import TEST_SYSTEM_DATA, TEST_SYSTEM_URL, USER_INPUT
+
+
+async def test_form(hass, aioclient_mock):
+    """Test that form shows up."""
+
+    aioclient_mock.get(
+        TEST_SYSTEM_URL,
+        text=TEST_SYSTEM_DATA,
+    )
+
+    result1 = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result1["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result1["step_id"] == "user"
+    assert result1["errors"] == {}
+
+    with patch(
+        "homeassistant.components.advantage_air.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result1["flow_id"],
+            USER_INPUT,
+        )
+
+    assert len(aioclient_mock.mock_calls) == 1
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "testname"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6319846)